### PR TITLE
Added redirect link to sendinblue

### PR DESCRIPTION
**Pull request by martinlagler**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| License | MIT

#### What's in this PR?

Adds redirect link to sendinblue.

### CODE REVIEW

File: Event/SendinblueListSubscriber.php [hunk 1]
```diff
@@ -36,12 +37,19 @@ class SendinblueListSubscriber implements EventSubscriberInterface
...
     private $contactsApi;
 
+    /**
+     * @var LinkProviderPoolInterface
+     */
+    private $linkProviderPool;
+
     public function __construct(
         RequestStack $requestStack,
+        LinkProviderPoolInterface $linkProviderPool,
```
Comment (Prokyonn) [inline Event/SendinblueListSubscriber.php]: @alexander-schranz can we do the BC break here or should we add a BC layer?

Comment (alexander-schranz) [reply Event/SendinblueListSubscriber.php]: as it is not much work a bc layer should be added here. includes that this need to be the the last parameter here and be ` = null` by default

File: Event/SendinblueListSubscriber.php [hunk 2]
```diff
@@ -126,4 +140,35 @@ public function listSubscribe(FormSavePostEvent $event): void
             $this->contactsApi->createDoiContact($createDoiContact);
         }
     }
+
+    /**
+     * @param array<string, ?string> $redirectLink
```
Comment (alexander-schranz) [inline Event/SendinblueListSubscriber.php]: define this more specific

File: Event/SendinblueListSubscriber.php [hunk 3]
```diff
@@ -97,6 +106,11 @@ public function listSubscribe(FormSavePostEvent $event): void
                 $mailTemplateId = $field['options']['mailTemplateId'] ?? null;
                 /** @var int|null $listId */
                 $listId = $field['options']['listId'] ?? null;
+                $redirectLink = $field['options']['redirectLink'] ?? null;
+
+                if ($redirectLink && $this->linkProviderPool) {
```
Comment (alexander-schranz) [inline Event/SendinblueListSubscriber.php]: Move check for `$this->linkProviderPool` that way phpstan now:

```suggestion
                if ($redirectLink) {
```

File: Event/SendinblueListSubscriber.php [hunk 4]
```diff
@@ -126,4 +140,44 @@ public function listSubscribe(FormSavePostEvent $event): void
...
+    {
+        if (!$redirectLink['provider']) {
+            return null;
+        }
+
+        if ('external' === $redirectLink['provider']) {
+            return $redirectLink['href'];
+        }
+
+        $linkProvider = $this->linkProviderPool->getProvider($redirectLink['provider']);
```
Comment (alexander-schranz) [inline Event/SendinblueListSubscriber.php:170]:
```suggestion
        if (!$this->linkProviderPool) {
            return null;
        }

        $linkProvider = $this->linkProviderPool->getProvider($redirectLink['provider']);
```

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

**Review by alexander-schranz — APPROVED** [review]

**Pull request merged**